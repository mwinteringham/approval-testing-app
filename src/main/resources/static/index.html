<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Approval testing practise app</title>
    <style>
        img {
            width: 400px;
        }

        .container {
            width: 400px;
            margin-left: auto;
            margin-right: auto;
            margin-top: 20px;
            border: 1px solid black;
            padding: 10px;
        }

        .detail {
            color: navy;
            text-align: right;
            float: right;
        }

        .buttonHolder {
            text-align: center;
        }

        button {
            font-size: 20px;
        }
    </style>
    <script src="./jquery.js"></script>
    <script>
        function guid() {
          function s4() {
            return Math.floor((1 + Math.random()) * 0x10000)
              .toString(16)
              .substring(1);
          }
          return s4() + s4() + '-' + s4() + '-' + s4() + '-' + s4() + '-' + s4() + s4() + s4();
        }

        $.get( "/car", function( data ) {
            $("#image").attr("src", data.image);
            $("#company").html(data.company);
            $("#make").html(data.make);
            $("#price").html(data.price);
            $("#purchase").html(data.datePurchased);

            $("#timestamp").html(Date.now());
            $("#guid").attr("value", guid());
        });

        var postToggle = function(){
            $.post( "/toggle", function( data ) {
                location.reload();
            });
        }
    </script>
</head>
<body>
    <div class="container">
        <div class="details">
            <img src="" id="image" />
            <h2>Company: <div class="detail" id="company">Unknown</div></h2>
            <h2>Make: <div class="detail" id="make">Unknown</div></h2>
            <h2>Price: <div class="detail" id="price">Unknown</div></h2>
            <h2>Purchase date: <div class="detail" id="purchase">Unknown</div></h2>
        </div>
        <p>Last updated: <span class="detail" id="timestamp"></span></p>
        <div class="buttonHolder">
            <button onclick="postToggle()">Toggle</button>
        </div>
        <input type="hidden" id="guid" value="" />
    </div>
</body>
</html>