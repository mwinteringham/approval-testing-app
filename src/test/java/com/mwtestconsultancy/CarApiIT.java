package com.mwtestconsultancy;

import io.restassured.response.Response;
import org.approvaltests.Approvals;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.springframework.boot.SpringApplication;
import org.springframework.context.ApplicationContext;

import static io.restassured.RestAssured.given;

public class CarApiIT {

    private ApplicationContext app;

    @Before
    public void setupApi(){
        app = SpringApplication.run(CarApi.class);
    }

    @After
    public void killApi(){
        SpringApplication.exit(app);
    }

    @Test
    public void testGetCar(){
        Response carResponse = given()
                                .get("http://localhost:8080/car");

        Approvals.verify(carResponse.getBody().print());
    }

    @Test
    public void testToggleCar(){
        Response toggleResponse = given()
                                    .post("http://localhost:8080/toggle");

        Approvals.verify(toggleResponse.getStatusCode());
    }

    @Test
    public void testToggleCarSendsNewPayload(){
        Response toggleResponse = given()
                .post("http://localhost:8080/toggle");

        Response carResponse = given()
                .get("http://localhost:8080/car");

        Approvals.verify(carResponse.getBody().print());
    }

}
