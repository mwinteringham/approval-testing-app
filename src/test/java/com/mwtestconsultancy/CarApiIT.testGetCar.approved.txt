{"company":"Ford","make":"Focus","price":20000,"datePurchased":"2018-01-01","image":"./focus.jpg"}
